### ISSUE TITLE
Use same version of jQuery on all of our sites

### ISSUE THREAD

**Issue by zoffixznet** [perl6/perl6.org#40]
Currently modules.perl6.org uses 2.1.4, but perl6.org uses an older version.

We need to sync up the two sites (and any other ones that use jQuery) to use the same version. Simplify maintenance.

**Comment by jonathanstowe** [comment]
Wouldn't a cdn hosted version help here?

**Comment by zoffixznet** [comment]
Sure, but we currently use 1.something CDN on perl6.org and newer 2.1.4 CDN on modules.perl6.org. We need to bring everything up to use same version.

I should have time to look at this tomorrow.

**Comment by moritz** [comment]
+1 to using the same jquery version everywhere, -1 to using a CDN. The privacy issues simply aren't worth it.

**Comment by zoffixznet** [comment]
> The privacy issues simply aren't worth it.

Would you have any info for me to look up regarding the privacy issues? I think it's the first time I hear of such an issue raised regarding serving JS libs from CDN.

**Comment by moritz** [comment]
@zoffixznet if you server JS files from a CDN, you give then one running the CDN a pretty accurate picture about who visits your site, and when. Which is really a leak of private information.

**Comment by coke** [comment]
Note that because of the nature of the request, the first request from a
user is the one the CDN gets; future requests will come from cache until
it's cleared.

I think that for a public site about an open source language where we
probably have volunteers donating serving costs, using a CDN makes sense.

On Mon, Oct 19, 2015 at 1:02 AM, Moritz Lenz <notifications@github.com>
wrote:

> @zoffixznet <https://github.com/zoffixznet> if you server JS files from a
> CDN, you give then one running the CDN a pretty accurate picture about who
> visits your site, and when. Which is really a leak of private information.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/perl6/perl6.org/issues/40#issuecomment-149099083>.
>



-- 
Will "Coke" Coleda

**Comment by asb-capfan** [comment]
I agree with @coke. Maybe it would be even better to let all the world know that the Perl6 web site is frequented across the word. It's kind of marketing.

**Comment by jonathanstowe** [comment]
I'm in the "not fussed by the potential privacy issue camp", but if it was considered to be important a fairly trivial expedient would be to make a rewriting proxy on the server(s) (trivial assuming apache with mod_rewrite available.)

**Comment by zoffixznet** [comment]
> make a rewriting proxy on the server(s) (trivial assuming apache with mod_rewrite available.) 

:+1: on that. And `mod_rewrite` can also be used for this Issue, if enabled: https://github.com/perl6/modules.perl6.org/issues/13

**Comment by jotasprout** [comment]
If this is still truly open, I'll put in the CDN code. I can't find any references to a jquery file in any of the files in my local repo.

**Comment by jotasprout** [comment]
I can't find a reference to a jquery file in any of the files in my local repo but looking at the source for the live site shows a link to a CDN. Should this issue be closed?

**Comment by zoffixznet** [comment]
The perl6.org's jQuery inclusion is [here](https://github.com/perl6/perl6.org/blob/d373b312e5d1eeffbd34817ef4954325f998bfa9/includes/footer#L71) and modules.perl6.org's jquery inclusion is [here](https://github.com/perl6/modules.perl6.org/blob/master/lib/ModulesPerl6.pm#L54). While the two versions are the same these days, it would be nice to see if they can be upgraded to latest and greatest. I see [jQuery 3.1](http://jquery.com/download/) is out, while the sites still use 1.13.

**Comment by zoffixznet** [comment]
Just remembered: our docs site [has jQuery on it](https://github.com/perl6/doc/blob/master/template/footer.html#L22) too. That one is version `1.11.3`

**Comment by coke** [comment]
Also #204
